lib/pushy_client/protocol_handler.rb

```diff
@@ -467,7 +459,7 @@ def send_signed_json_command(method, json)
       @socket_lock.synchronize do
         @command_socket_outgoing_seq += 1
         json[:sequence] = @command_socket_outgoing_seq
-        message = JSON.generate(json)
+        message = JSON.generate(validate_params(json))
```

Comment:
We can't use validate_params as is here unfortunately. We need to measure the size of the JSON.generate output directly to know the size, and so calling it *before* JSON.generate won't work. We'll probably have to JSON.generate, measure size. If the packet is too large,  adjust the packet to fit and rerun JSON.generate.